File: lecture07/03_subsequence/template.py
#!/usr/bin/env python3

import sys

# Functions

def is_subsequence(s, t):
    # TODO: determine if s is a subsequence of t

def main():
    for line in sys.stdin:
        s, t = map(list, line.split())
        print(is_subsequence(s, t))

# Main Execution

if __name__ == '__main__':
    main()

File: lecture07/03_subsequence/template.cpp
// subsequence.cpp

#include <iostream>
#include <string>

using namespace std;

// Functions

bool is_subsequence(string &s, string &t) {
    // TODO: determine if s is a subsequence of t
}

// Main execution

int main(int argc, char *argv[]) {
    string s, t;

    while (cin >> s >> t) {
    	cout << (is_subsequence(s, t) ? "True" : "False") << endl;
    }

    return 0;
}
